### PR TITLE
expression: rewrite builtin string function: BIN

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

File: expression/integration_test.go [hunk 1]
```diff
@@ -711,6 +711,13 @@ func (s *testIntegrationSuite) TestBuiltin(c *C) {
 	result = tk.MustQuery("select cast(1 as signed int)")
 	result.Check(testkit.Rows("1"))
 
+	result = tk.MustQuery(`select bin(-1);`)
```
Comment (XuHuaiyu) [inline expression/integration_test.go]: Move these test to TestStringBuiltin

File: expression/builtin_string_test.go [hunk 2]
```diff
@@ -1702,6 +1702,8 @@ func (s *testEvaluatorSuite) TestBin(c *C) {
 	for _, t := range dtbl {
 		f, err := fc.getFunction(datumsToConstants(t["Input"]), ctx)
 		c.Assert(err, IsNil)
+		c.Assert(f, NotNil)
```
Comment (XuHuaiyu) [inline expression/builtin_string_test.go]: why check f notnil here?

Comment (zz-jason) [inline expression/builtin_string_test.go]: to guarantee the following dereference of `f` not cause panic

File: expression/integration_test.go (hunk 1 above)
Comment (zimulala) [inline expression/integration_test.go]: PTAL @zz-jason

**Comment by zimulala** [comment]
Rest LGTM

**Comment by zz-jason** [comment]
updated, @zimulala @XuHuaiyu PTAL again

**Comment by zimulala** [comment]
LGTM

**Pull request merged**